app/qml/MerginSettingDialog.qml

```diff
@@ -0,0 +1,153 @@
+import QtQuick.Dialogs 1.2
+import QtGraphicalEffects 1.0
+import QgsQuick 0.1 as QgsQuick
+
+Dialog {
+    property string linkFieldText: __merginApi.apiRoot
+    property real fieldHeight: InputStyle.rowHeight
+    property color fontColor: InputStyle.fontColor
+    property real panelMargin: InputStyle.panelMargin
+
+    signal closing()
+
+    id: merginLinkDialog
+    modality: Qt.WindowModal
+    title: "Set Mergin instance"
+
+    onApply: {
+        __merginApi.apiRoot = linkField.text
+        merginLinkDialog.close()
+        closing()
+    }
+    onRejected: {
+        linkField.text = __merginApi.apiRoot
+        merginLinkDialog.close()
+        closing()
+    }
+    onReset: {
+        __merginApi.resetApiRoot()
+        merginLinkDialog.close()
+        closing()
+    }
+
+
+    contentItem: Rectangle {
+        id: dialogContent
+        width: window.width - 2 * merginLinkDialog.panelMargin
+        height: fieldHeight * 4
+
+        Keys.onEnterPressed: merginLinkDialog.accept()
+        Keys.onEscapePressed: merginLinkDialog.reject()
+        Keys.onBackPressed: merginLinkDialog.reject()
+        color: "white"
+
+        Item {
+            anchors.fill: parent
+            anchors.margins: merginLinkDialog.panelMargin
+
+            ColumnLayout {
+                width: parent.width
+                height: parent.height
+                anchors.margins: merginLinkDialog.panelMargin
+
+                RowLayout {
+                    id: row
+                    Layout.fillWidth: true
+                    height: fieldHeight
+
+                    Rectangle {
+                        id: iconContainer
+                        height: fieldHeight/2
+                        width: fieldHeight/2
+                        color: "white"
+
+                        Image {
+                            anchors.margins: (fieldHeight/4)
+                            id: icon
+                            height: iconContainer.height
+                            width: height
+                            source: 'link.svg'
+                            sourceSize.width: width
+                            sourceSize.height: height
+                            fillMode: Image.PreserveAspectFit
+                        }
+
+                        ColorOverlay {
+                            anchors.fill: icon
+                            source: icon
+                            color: merginLinkDialog.fontColor
+                        }
+                    }
+
+                    TextField {
+                        id: linkField
+                        text: merginLinkDialog.linkFieldText
+                        height: fieldHeight
+                        font.pixelSize: InputStyle.fontPixelSizeNormal
+                        color: merginLinkDialog.fontColor
+                        placeholderText: qsTr("Mergin instance")
+                        inputMethodHints: Qt.ImhNoPredictiveText | Qt.ImhNoAutoUppercase
```

Comment:
Unrelated, but can we have this hint also for username in auth panel? I have seen that on mobile it tries to start with upper case